jlama-core/src/main/java/com/github/tjake/jlama/tensor/KvBufferCache.java

```diff
@@ -52,7 +54,19 @@ public KvBufferCache(AbstractModel model) {
     }
 
     public KvBuffer getKvBuffer(UUID session) {
-        return kvBufferCache.computeIfAbsent(session, s -> new KvBuffer(s, 1 << 24)); // 16MB per page
+        return kvBufferCache.computeIfAbsent(session, s -> new KvBuffer(s, 1 << 23, false)); // 8MB per page
+    }
+
+    public KvBuffer getEphemeralKvBuffer() {
+        return new KvBuffer(UUID.randomUUID(), 1 << 20, true);
+    }
+
+    @Override
+    public void close() {
+        for (KvBuffer kvBuffer : kvBufferCache.values()) {
+            kvBuffer.close();
+        }
+        kvBufferCache.clear();
```

Comment:
We can improve this by iterating over kvBufferCache only once:
```
for (Iterator<Map.Entry<UUID, KvBuffer>> it = kvBufferCache.entrySet().iterator(); it.hasNext();) {
    it.next().getValue().close();
    it.remove();
}
```

https://stackoverflow.com/questions/1884889/iterating-over-and-removing-from-a-map/29187813#29187813